src/drivers/driver.ts

```diff
@@ -169,69 +169,56 @@ export abstract class CMakeDriver implements vscode.Disposable {
 
   /**
    * The environment variables required by the current kit
+   * The ${dollar} are not expanded and missing environment variable subs are not expanded for this variables
    */
-  private _kitEnvironmentVariables = new Map<string, string>();
-
-  /**
-   * Compute the environment variables that apply with substitutions by expansionOptions
-   */
-  async computeExpandedEnvironment(in_env: proc.EnvironmentVariables, expanded_env: proc.EnvironmentVariables): Promise<proc.EnvironmentVariables> {
-    const env = {} as {[key: string]: string};
-    const opts = this.expansionOptions;
-
-    await Promise.resolve(
-      util.objectPairs(in_env)
-      .forEach(async ([key, value]) => env[key] = await expand.expandString(value, {...opts, envOverride: expanded_env}))
-    );
-    return env;
-  }
+  private _kitEnvironmentVariables: proc.EnvironmentVariables = {};
 
   /**
    * Get the environment variables that should be set at CMake-configure time.
    */
   async getConfigureEnvironment(): Promise<proc.EnvironmentVariables> {
+    const envs_to_merge: (proc.EnvironmentVariables | undefined)[] = [];
     if (this.useCMakePresets) {
-      return this._configurePreset?.environment as proc.EnvironmentVariables;
+      envs_to_merge.push(this._configurePreset?.environment as proc.EnvironmentVariables);
     } else {
-      let envs = getKitEnvironmentVariablesObject(this._kitEnvironmentVariables);
-      /* NOTE: By mergeEnvironment one by one to enable expanding self containd variable such as PATH properly */
-      /* If configureEnvironment and environment both configured different PATH, doing this will preserve them all */
-      envs = util.mergeEnvironment(envs, await this.computeExpandedEnvironment(this.config.environment, envs));
-      envs = util.mergeEnvironment(envs, await this.computeExpandedEnvironment(this.config.configureEnvironment, envs));
-      envs = util.mergeEnvironment(envs, await this.computeExpandedEnvironment(this._variantEnv, envs));
-      return envs;
+      envs_to_merge.push(this._kitEnvironmentVariables);
     }
+    envs_to_merge.push(this.config.environment);
+    envs_to_merge.push(this.config.configureEnvironment);
+    envs_to_merge.push(this._variantEnv);
+    return expand.mergeEnvironmentWithExpand(false, envs_to_merge, this.expansionOptions);
   }
 
   /**
    * Get the environment variables that should be set at CMake-build time.
    */
   async getCMakeBuildCommandEnvironment(in_env: proc.EnvironmentVariables): Promise<proc.EnvironmentVariables> {
-    let envs: proc.EnvironmentVariables;
+    const envs_to_merge: (proc.EnvironmentVariables | undefined)[] = [in_env];
```

Comment:
Thank you for your contribution. 
Would you please provide some explanation on why `mergeEnvironmentWithExpand` is needed here?
There is a `mergeEnvironment`  and `splitEnvironmentVars` and `computeExpandedEnvironment`, and it is a little confusing having another function. We can add more comments/samples like what you did for `mergeEnvironmentWithExpand` to add some clarity, or choose some more clear names.

Reply:
```ts

/**
 * mergeEnvironmentWithExpand will merge new env variable and
 * expand it one by one, so that the environment like this
 * will expand properly:
 * config.environment
 * { PATH:`${env.PATH};C:\\MinGW\\bin` }
 * config.configureEnvironment
 * { PATH:`${env.PATH};C:\\OtherPath\\bin` }
 *
 * @param intermediateExpand If this is the intermediate expanding
 * procedure or the final expanding procedure, if it's the
 * intermediate expanding procedure, the variable that not found
 * won't not expaned, if it's the final expanding procedure, the variable
 * that not found would expand to ''.
 * @param envs The list of environment variables to expand
 * @param opts Environment expand options
 *
 * NOTE: By mergeEnvironment one by one to enable expanding self
 * containd variable such as PATH properly. If configureEnvironment
 * and environment both configured different PATH, doing this will
 * preserve them all
 */
export async function mergeEnvironmentWithExpand(
  intermediateExpand: boolean,
  envs: (EnvironmentVariables | undefined)[],
  opts?: ExpansionOptions)
```
I've document it in mergeEnvironmentWithExpand, don't know if this is proper

Reply:
I appreciate if you could add some comments/samples to `mergeEnvironment` and `splitEnvironmentVars` and `computeExpandedEnvironment`, similar to what you did for `mergeEnvironmentWithExpand` to add some clarity.

Reply:
I didn't touch and use of splitEnvironmentVars, other function are documented